python/kserve/kserve/protocol/rest/openai/errors.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2024 The KServe Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from http import HTTPStatus
+from fastapi.responses import JSONResponse
+from kserve.logging import logger
+
+
+class OpenAIError(Exception):
+    """
+    Exception class for generic OpenAI error.
+    """
+
+    def __init__(self, reason):
+        self.reason = reason
+
+    def __str__(self):
+        return self.reason
+
+
+async def openai_error_handler(_, exc):
+    logger.error("Exception:", exc_info=exc)
+    return JSONResponse(
```

Comment:
I think we need to return the openai defined Error type, otherwise it will not be compatible with openai client?

Reply:
Sending contents of type OpenAI error object

For error scenarios the body looks like this
`{'code': '500', 'message': 'Either prompt or prompt_ids should be provided.', 'param': '', 'type': 'OpenAIError'}`